modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/HistoryManager.kt

```diff
@@ -0,0 +1,89 @@
+package au.com.shiftyjelly.pocketcasts.repositories.podcast
+
+import au.com.shiftyjelly.pocketcasts.models.entity.Episode
+import au.com.shiftyjelly.pocketcasts.models.to.HistorySyncResponse
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.utils.extensions.parseIsoDate
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.rx2.awaitSingleOrNull
+import kotlinx.coroutines.withContext
+import java.util.Date
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+class HistoryManager @Inject constructor(
+    private val podcastManager: PodcastManager,
+    private val episodeManager: EpisodeManager,
+    private val settings: Settings,
+) : CoroutineScope {
+
+    companion object {
+        const val ACTION_ADD = 1
+        const val ACTION_DELETE = 2
+    }
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    /**
+     * Read the server listening history response.
+     * @param response The server response.
+     * @param updateServerModified Set to true when this is latest listening history, rather than part of the user's history.
+     */
+    suspend fun processServerResponse(response: HistorySyncResponse, updateServerModified: Boolean) = withContext(Dispatchers.IO) {
+        if (!response.hasChanged(0) || response.changes.isNullOrEmpty()) {
```

Comment:
I moved this logic from the SyncHistoryTask so the End of Year logic could share it.